File: flask-backend/seed.py
"""
seed.py — Database seeding script for development & demo deployment.

Populates initial Admin and Member accounts, along with sample leads,
notes, and activity logs.

Usage:
  python seed.py
"""
import os
from app import create_app, db
from app.models.user import User
from app.models.lead import Lead
from app.models.note import Note
from app.models.activity_log import ActivityLog
from app.services.activity_service import ActivityService

app = create_app(os.environ.get("FLASK_ENV", "development"))

def seed():
    with app.app_context():
        print("🌱 Resetting and seeding database...")
        db.drop_all()
        db.create_all()

        # Admin user
        admin = User(name="Admin User", email="admin@leadflow.com", role="admin")
        admin.set_password("password123")
        db.session.add(admin)

        # Member users (Sales Reps)
        member1 = User(name="John SalesRep", email="john@leadflow.com", role="member")
        member1.set_password("password123")
        db.session.add(member1)

        member2 = User(name="Sarah Closer", email="sarah@leadflow.com", role="member")
        member2.set_password("password123")
        db.session.add(member2)

        db.session.commit()
        print("✅ Created default users: admin@leadflow.com, john@leadflow.com, sarah@leadflow.com (password: password123)")

        # Sample Leads
        lead1 = Lead(
            name="Acme Corp Lead",
            email="procurement@acmecorp.com",
            phone="+1 (555) 019-2831",
            company="Acme Corp",
            source="Website",
            message="Looking for enterprise CRM solution for 50 sales reps.",
            status="Proposal Sent",
            assigned_to_id=member1.id,
            created_by_id=admin.id,
        )
        lead2 = Lead(
            name="Starlight Tech Inbound",
            email="cto@starlight.io",
            phone="+1 (555) 982-1122",
            company="Starlight Tech",
            source="Website",
            message="Interested in custom RBAC and API integration features.",
            status="Qualified",
            assigned_to_id=member2.id,
            created_by_id=admin.id,
        )
        lead3 = Lead(
            name="Globex Prospect",
            email="info@globex.com",
            phone="+1 (555) 443-8821",
            company="Globex Corp",
            source="Website",
            message="Requesting product demo.",
            status="New",
            assigned_to_id=None,
            created_by_id=None,
        )

        db.session.add_all([lead1, lead2, lead3])
        db.session.commit()

        # Add notes & activity logs
        note1 = Note(lead_id=lead1.id, author_id=member1.id, body="Sent customized proposal PDF via email. Awaiting response.")
        db.session.add(note1)

        ActivityService.log(lead1.id, 'Status updated to "Proposal Sent"', actor_id=member1.id)
        ActivityService.log(lead1.id, f"Lead assigned to {member1.name}", actor_id=admin.id)
        ActivityService.log(lead2.id, 'Status updated to "Qualified"', actor_id=member2.id)

        db.session.commit()
        print("🚀 Database seeding completed successfully!")

if __name__ == "__main__":
    seed()

File: backend/seed.js
require('dotenv').config();
const mongoose = require('mongoose');
const User = require('./models/User');
const Lead = require('./models/Lead');
const Note = require('./models/Note');
const Activity = require('./models/Activity');

/**
 * Exported for programmatic use (e.g., auto-seed from server.js)
 * Assumes DB is already connected when called as a module.
 * When called directly as a script, connects & disconnects itself.
 */
const seedDatabase = async () => {
  // Clear existing data
  await Promise.all([
    User.deleteMany({}),
    Lead.deleteMany({}),
    Note.deleteMany({}),
    Activity.deleteMany({}),
  ]);

  // Create Admin and Member users
  const admin = await User.create({
    name: 'Admin User',
    email: 'admin@leadflow.com',
    password: 'password123',
    role: 'Admin',
  });

  const member1 = await User.create({
    name: 'John SalesRep',
    email: 'john@leadflow.com',
    password: 'password123',
    role: 'Member',
  });

  const member2 = await User.create({
    name: 'Sarah Closers',
    email: 'sarah@leadflow.com',
    password: 'password123',
    role: 'Member',
  });

  console.log('✅ Created users: admin@leadflow.com, john@leadflow.com, sarah@leadflow.com');

  // Create sample leads
  const leads = await Lead.create([
    {
      name: 'Acme Enterprise Deal',
      email: 'procurement@acmecorp.com',
      phone: '+1 (555) 234-5678',
      company: 'Acme Corp',
      message: 'Looking for enterprise CRM solution for 50 sales reps.',
      status: 'Proposal Sent',
      assignedTo: member1._id,
      createdBy: admin._id,
      source: 'Website',
    },
    {
      name: 'Starlight Tech Inc',
      email: 'cto@starlight.io',
      phone: '+1 (555) 987-6543',
      company: 'Starlight Tech',
      message: 'Interested in API integrations and custom RBAC features.',
      status: 'Qualified',
      assignedTo: member2._id,
      createdBy: admin._id,
      source: 'Website',
    },
    {
      name: 'Globex Corp Inbound',
      email: 'info@globex.com',
      phone: '+1 (555) 345-6789',
      company: 'Globex Corp',
      message: 'Requesting demo of lead distribution features.',
      status: 'New',
      assignedTo: null,
      createdBy: null,
      source: 'Website',
    },
    {
      name: 'Apex Digital Agency',
      email: 'alex@apexdigital.com',
      phone: '+1 (555) 876-5432',
      company: 'Apex Digital',
      message: 'Signed contract for annual plan!',
      status: 'Won',
      assignedTo: member1._id,
      createdBy: admin._id,
      source: 'Manual',
    },
  ]);

  console.log(`✅ Created ${leads.length} sample leads.`);

  // Add initial activity logs & notes
  await Activity.create({
    leadId: leads[0]._id,
    action: 'Lead status changed to "Proposal Sent"',
    performedBy: member1._id,
  });

  await Note.create({
    leadId: leads[0]._id,
    userId: member1._id,
    text: 'Sent customized proposal PDF via email. Awaiting feedback from CFO.',
  });

  console.log('🚀 Seeding completed successfully!');
};

module.exports = { seedDatabase };

// Allow running directly as a script: node seed.js
if (require.main === module) {
  const connectDB = require('./config/db');
  connectDB()
    .then(() => seedDatabase())
    .then(() => process.exit(0))
    .catch((err) => {
      console.error('❌ Seeding failed:', err.message);
      process.exit(1);
    });
}
